app/javascript/lib/imperial.js

```diff
@@ -1277,110 +1277,36 @@ export default class Imperial {
       }
       case 'taxation': {
         const nationName = action.payload.nation;
-        const nation = this.nations.get(nationName);
-        const previousTaxChartPosition = nation.taxChartPosition;
+
         // 1. Tax revenue / success bonus
         const taxes = this.getTaxes(nationName);
-        nation.taxChartPosition = this.getTaxChartPosition(taxes);
-        let excessTaxes = nation.taxChartPosition - previousTaxChartPosition;
-        // Players can never lose money here
-        if (excessTaxes < 0) {
-          excessTaxes = 0;
-        }
-        // Player receives full excess taxes
-        this.players[this.currentPlayerName].cash += excessTaxes;
+        const bonus = this.playerBonusAfterUnitMaintenanceCosts(nationName, taxes);
+
+        this.players[this.currentPlayerName].cash += bonus;
         this.annotatedLog.push(
           Action.playerGainsCash({
             player: this.currentPlayerName,
-            amount: excessTaxes,
```

Comment:
Also fixes the double logging of tax bonuses for `Imperial 2030` & `Imperial Asia`
![image](https://user-images.githubusercontent.com/28380984/226119079-d2a59fde-aea9-468a-8e4c-e4898ad64a41.png)